crates/sncast/src/lib.rs

```diff
@@ -86,6 +88,20 @@ pub fn get_provider(url: &str) -> Result<JsonRpcClient<HttpTransport>> {
     Ok(provider)
 }
 
+pub async fn warn_if_incompatible_rpc_version(
+    provider: &JsonRpcClient<HttpTransport>,
+    url: &str,
+) -> Result<()> {
+    let node_spec_version = get_and_parse_spec_version(provider).await?;
+    if !is_supported_version(&node_spec_version) {
+        print_as_warning(&anyhow!(
+            "The RPC node with the url {url} is using an unsupported version {node_spec_version}. The current version supported by sncast is {EXPECTED_RPC_VERSION}"
```

Comment:
I wonder if we should print that warning if the major version is correct. Especially considering there might be just minors difference between versions returned by the node e.g. `0.5.0rc4`, `0.5.0` and `0.5.1`

Reply:
I guess that nodes would return `0.5.0` even for cases like `0.5.0rc4`, but I am not sure. Outputting that warning seems reasonable even when the difference is only in the minor version, as it can still result in unpredictable behavior.

Reply:
> I guess that nodes would return `0.5.0` even for cases like `0.5.0rc4`, but I am not sure. 

Not really, they sometimes do output rc version.

> Outputting that warning seems reasonable even when the difference is only in the minor version, as it can still result in unpredictable behavior.

Maybe you're right, just that warning is not stating that version "might be incompatible", but that it's "unsupported", which I'm not sure is often correct for minor versions

Reply:
Agree with Maksim here

Reply:
I'd change the wording from `unsupported` to `incompatible` as well, tbh. 
But I'd still display it for minor, patch and rc versions, because (and I assume you meant `minor` and not `major`, as major has always been `0`, and will be just that for the forseeable future):

- generally rc versions should not make their way to nodes used by "normal" people (these are rcs, so should rather be used for test purposes only ad bugs are expected)
- sw are rather relaxed about their semver, and they also introduce braking changes to patch versions. so our check would be worthless in that case - I'd rather be more explicit.